src/components/views/messages/SenderProfile.js

```diff
@@ -0,0 +1,43 @@
+/*
+ Copyright 2015, 2016 OpenMarket Ltd
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+'use strict';
+
+import React from 'react';
+import {emojifyText} from '../../../HtmlUtils';
+
+export default function SenderProfile(props) {
+    const {mxEvent} = props;
+    const name = mxEvent.sender ? mxEvent.sender.name : mxEvent.getSender();
+    const {msgtype} = mxEvent.getContent();
+
+    if (msgtype === 'm.emote') {
+        return <span />; // emote message must include the name so don't duplicate it
+    }
+
+    return (
+        <span className="mx_SenderProfile"
+              dangerouslySetInnerHTML={emojifyText(`${name || ''} ${props.aux || ''}`)}
+              onClick={props.onClick}>
+        </span>
+    );
+}
+
+SenderProfile.propTypes = {
+    mxEvent: React.PropTypes.object.isRequired, // event whose sender we're showing
+    aux: React.PropTypes.string, // stuff to go after the sender name, if anything
+    onClick: React.PropTypes.func,
+};
```

Comment:
The emojification code is in matrix-react-sdk.

On Tue 9 Aug, 2016, 10:07 PM David Baker, <notifications@github.com> wrote:

> In src/components/views/messages/SenderProfile.js
> <https://github.com/matrix-org/matrix-react-sdk/pull/395#discussion_r74095226>
> :
>
> > +        return <span />; // emote message must include the name so don't duplicate it
> > +    }
> > +
> > +    return (
> > +        <span className="mx_SenderProfile"
> > +              dangerouslySetInnerHTML={emojifyText(`${name || ''} ${props.aux || ''}`)}
> > +              onClick={props.onClick}>
> > +        </span>
> > +    );
> > +}
> > +
> > +SenderProfile.propTypes = {
> > +    mxEvent: React.PropTypes.object.isRequired, // event whose sender we're showing
> > +    aux: React.PropTypes.string, // stuff to go after the sender name, if anything
> > +    onClick: React.PropTypes.func,
> > +};
>
> What's the reason for moving this from vector?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/matrix-org/matrix-react-sdk/pull/395/files/3aa3530d998fdb42e6c94b3239d80a2d1adfc3a2#r74095226>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOlRLH43aHgoUsRGfItT5Ur3wrQfB4gks5qeKy6gaJpZM4JgQie>
> .
>